OsmAnd/src/net/osmand/plus/plugins/development/OsmandDevelopmentPlugin.java

```diff
@@ -262,4 +279,89 @@ public OrderedLineDataSet getOrderedLineDataSet(@NonNull LineChart chart, @NonNu
 	protected TrackPointsAnalyser getTrackPointsAnalyser() {
 		return AutoZoomBySpeedHelper.getTrackPointsAnalyser(app);
 	}
-}
\ No newline at end of file
+
+	private boolean fpsStatsEnabled = false;
+	private final int FPS_STATS_INTERVAL_SECONDS = 1; // 10 ?
+	private final int FPS_STATS_LIFETIME_MINUTES = 15;
+	private Handler fpsStatsHandler = new Handler(Looper.getMainLooper());
+	private List<FpsStatsEntry> fpsStats = Collections.synchronizedList(new ArrayList<>());
+
+	public class FpsStatsEntry {
+		public long timestamp;
+		public float battery;
+		public float fps1k;
+		public float idle1k;
+		public float gpu1k;
+
+		public FpsStatsEntry(OsmandApplication app) {
+			MapRendererView renderer = app.getOsmandMap().getMapView().getMapRenderer();
+			if (renderer != null) {
+				this.timestamp = System.currentTimeMillis();
+
+				this.fps1k = renderer.getFrameRateLast1K();
+				this.idle1k = renderer.getIdleTimePartLast1K();
+				this.gpu1k = renderer.getGPUWaitTimePartLast1K();
+
+				Intent batteryIntent = app.registerReceiver(null, new IntentFilter(Intent.ACTION_BATTERY_CHANGED));
+				int level = batteryIntent.getIntExtra(BatteryManager.EXTRA_LEVEL, -1);
+				int scale = batteryIntent.getIntExtra(BatteryManager.EXTRA_SCALE, -1);
+				this.battery = level != -1 && scale != -1 && scale > 0 ? level * 100 / scale : 0;
+			}
+		}
+
+		public FpsStatsEntry(List<FpsStatsEntry> list, int minutes) {
+			if (!list.isEmpty()) {
+				this.battery = minuteBatteryUsage(list, minutes);
+				this.fps1k = avgFloat(list, minutes, entry -> entry.fps1k);
+				this.gpu1k = avgFloat(list, minutes, entry -> entry.gpu1k);
+				this.idle1k = avgFloat(list, minutes, entry -> entry.idle1k);
+			}
+		}
+
+		private float avgFloat(List<FpsStatsEntry> list, int minutes, Function<FpsStatsEntry, Float> get) {
```

Comment:
parameters should not have abstract names like `list`

Reply:
renamed list, minutes, etc